calblueprint/meepanyar
# [Inventory] Add Inventory Updates functionality

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
- Added basic functionality to create an inventory update
- Trigger an inventory update when inventory is added to a site (to track the user id and other details for recordkeeping)
- Added calculation for the `lastUpdated` timestamp based on the most recent purchase request OR inventory update
- Updated the way we calculate and format date/time strings to be sensitive to location and time zone.

## How to review

**Add inventory**
1. On the main inventory screen, select the + button to add a new inventory item
2. Select any item, input a quantity, and submit.
3. Check both Airtable and Redux to see that an Inventory Update was created for the new item.

**Create an inventory update**
1. Select any inventory item from the main list.
2. Click **Update**
3. Enter any quantity (note: we don't have robust edge case handling, so no need to try to break this) and click Update
4. Check that the quantity in the inventory profile is updated, as well as in the inventory record, and that an inventory update was created (check in both Airtable and Redux).
5. Check that a new InventoryUpdateCard shows up on the profile page at the **top** representing your most recent update.
6. Check that the Last Updated date reflects the approximate time you hit submit.

**Last Updated Date**
1. On an inventory item that already has an inventory update (basically all of them), create a purchase request.
2. Make sure the 'Last Updated' date now reflects the time you submitted the purchase request.

**Test Redirect**
1. Reset the Redux store in the inspector. You may also have to do Clear site data in the application tab as well.
2. Log in as testUser
3. Directly enter http://localhost:3000/inventory/updates/create into the URL bar
4. Make sure it redirects to the main inventory screen

[//]: # 'The order in which to review files and what to expect when testing locally'

## Relevant Links

### Online sources
https://momentjs.com/docs/#/get-set/max/
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString

### Related PRs
n/a

## Next steps
- Handle error states and edge case inputs
- Update frontend to reflect the designs

### Screenshots
![image](https://user-images.githubusercontent.com/21160510/113358451-ac972d00-92fa-11eb-9f26-c16fe7f86e1f.png)
![image](https://user-images.githubusercontent.com/21160510/113358477-b6209500-92fa-11eb-9496-2f915e6f03e0.png)

CC: @julianrkung

[//]: # 'This tags Julian as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
